pkg/controller/control.go

```diff
@@ -1170,6 +1177,86 @@ func (c *Controller) GetLatestMetics() *ptypes.Metrics {
 	return metrics
 }
 
+func (c *Controller) ShowProfiler() (string, error) {
+	c.profilerLock.Lock()
+	defer c.profilerLock.Unlock()
+
+	logrus.Infof("Prepare to do ShowProfilerConfig")
+	if c.profilerServer != nil {
+		return c.profilerServer.Addr, nil
+	}
+	return "", nil
+}
+
+func (c *Controller) EnableProfiler(portNumber int32) (string, error) {
+	c.profilerLock.Lock()
+	defer c.profilerLock.Unlock()
+
+	logrus.Infof("Prepare to do EnableProfiler")
+	var err error
+	if c.profilerServer != nil {
+		runningPort := strings.Split(c.profilerServer.Addr, ":")[1]
+		if runningPort == strconv.Itoa(int(portNumber)) {
+			return "", fmt.Errorf("profiler server is already running at %v", c.profilerServer.Addr)
+		}
+		return c.profilerServer.Addr, nil
+	}
+	// If we specific portNumber, use it as the port of profiler
+	// grpcProt is unique, we could use it as the base port of profiler
+	profilerPort := int(portNumber)
+	if profilerPort == 0 {
+		var grpcPort int
+		grpcPort, err = strconv.Atoi(strings.Split(c.GRPCAddress, ":")[1])
+		if err != nil {
+			logrus.Warnf("Failed to convert the gRPC port to integer.")
+			return "", err
+		}
+		profilerPort = types.ProfilerPortBase + grpcPort
+	}
+	profilerAddr := fmt.Sprintf(":%d", profilerPort)
+	c.profilerServer = &http.Server{
+		Addr:              profilerAddr,
+		ReadHeaderTimeout: 10 * time.Second,
+	}
+	logrus.Infof("Profiler of volume %v is enabled at %v", c.VolumeName, profilerAddr)
+	go func() {
+		if err := c.profilerServer.ListenAndServe(); err != http.ErrServerClosed {
+			logrus.Warnf("Get error when start profiler server(%v): %v.", c.profilerServer.Addr, err)
+			c.profilerServer = nil
+			return
+		}
+		logrus.Infof("Profiler server (%v) is closed.", c.profilerServer.Addr)
+	}()
+	logrus.Infof("Waiting the profiler server(%v) to start.", c.profilerServer.Addr)
+	// Wait for the profiler server to start, and check the profiler server.
+	time.Sleep(3 * time.Second)
+	if c.profilerServer == nil {
+		return "", fmt.Errorf("failed to start profiler server(%v)", profilerAddr)
+	}
+	return c.profilerServer.Addr, nil
```

Comment:
This waiting seems best effort to start the profiler. 

Can we just update the error into `profilerServer` if it can't be started, so we can simply know if the profiler is really running asynchronously by the show method. If it's not running, the caller can re-enable it.